server/mailrepository/mailrepository-api/src/main/java/org/apache/james/mailrepository/api/MailRepositoryUrl.java

```diff
@@ -39,21 +39,28 @@ public static final MailRepositoryUrl from(String url) {
     }
 
     private final String value;
+    private final String path;
     private final Protocol protocol;
 
     private MailRepositoryUrl(String value) {
         Preconditions.checkNotNull(value);
         Preconditions.checkArgument(value.contains(":"), "':' is mandatory to delimit protocol");
         this.value = value;
-        this.protocol = new Protocol(Splitter.on(':')
+        String protocol = Splitter.on(':')
             .splitToList(value)
-            .get(0));
+            .get(0);
+        this.protocol = new Protocol(protocol);
+        this.path = value.substring(protocol.length() + 3);
```

Comment:
`3` looks like a magic value. And nothing guarantees you'll have "://" at the right place

Reply:
Well, I thought the format was: `protocol + "://" + path`, is there methods to extract it?

Reply:
Yes but you need to be defensive about that and check your input